### PR TITLE
Stringification of PTX to avoid file deployment

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: plugins/engines/optix/cuda/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@
 # ".mic"-suffix trick, so we'll put libraries into separate
 # directories (names 'intel64' and 'mic', respectively)
 
+include(StringifyPtx)
+
+set(CUDA_PTX_DIR ${CMAKE_BINARY_DIR}/plugins/engines/optix/cuda)
```
Comment (rdumusc) [inline plugins/engines/optix/cuda/CMakeLists.txt]: unused, remove

File: plugins/engines/optix/cuda/CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +23,16 @@ set(BRAYNSOPTIXCUDAPLUGIN_SOURCES
   Constantbg.cu
 )
 
+set(BRAYNSOPTIXCUDAPLUGIN_PTX
+    braynsOptiXCudaPlugin_generated_Camera.cu.ptx
```
Comment (rdumusc) [inline plugins/engines/optix/cuda/CMakeLists.txt]: indentation (2 spaces)

File: plugins/engines/optix/cuda/CMakeLists.txt [hunk 3]
```diff
@@ -62,6 +61,13 @@ CUDA_WRAP_SRCS(
   OPTIONS ${options}
 )
 
+# Stringify PTX files and add generated files to source and header files
+if(BRAYNSOPTIXCUDAPLUGIN_PTX)
+    stringify_ptx(${BRAYNSOPTIXCUDAPLUGIN_PTX})
```
Comment (rdumusc) [inline plugins/engines/optix/cuda/CMakeLists.txt]: indentation

File: CMake/StringifyPtx.cmake [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2015-2016, EPFL/Blue Brain Project
+# All rights reserved. Do not distribute without permission.
+# Responsible Author: Raphael Dumusc <raphael.dumusc@epfl.ch>
+#
+# This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+
+set(CMAKE_STRINGIFY_PTX ${CMAKE_CURRENT_LIST_DIR})
```
Comment (favreau) [inline CMake/StringifyPtx.cmake]: Done

File: CMake/StringifyPtx.cmake [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    add_custom_command(OUTPUT ${_output_header} ${_output_source}
+      COMMAND ${CMAKE_COMMAND} -DSTRINGIFY_PTX_PROCESSING_MODE=ON
+        -DINPUT="${_input}" -DOUTPUT="${_output}"
+        -P ${CMAKE_STRINGIFY_PTX}/StringifyPtx.cmake
+        DEPENDS ${_input}
+      )
+
+    list(APPEND _output_headers ${_output_header})
+    list(APPEND _output_sources ${_output_source})
+  endforeach(_file ${ARGN})
```
Comment (favreau) [inline CMake/StringifyPtx.cmake]: Ok. Done

File: CMake/StringifyPtx.cmake [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  set(PTX_HEADERS ${_output_headers} PARENT_SCOPE)
+  set(PTX_SOURCES ${_output_sources} PARENT_SCOPE)
+endfunction()
+
+if(STRINGIFY_PTX_PROCESSING_MODE)
+  get_filename_component(FILENAME ${INPUT} NAME)
```
Comment (favreau) [inline CMake/StringifyPtx.cmake]: Done

File: CMake/StringifyPtx.cmake [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  string(REGEX REPLACE "[.]" "_" NAME ${FILENAME})
```
Comment (favreau) [inline CMake/StringifyPtx.cmake]: Done

File: CMake/StringifyPtx.cmake [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    "extern char const* const ${NAME};\n"
+    )
+
+  file(WRITE ${OUTPUT}.cpp
+    "/* Generated file, do not edit! */\n\n"
+    "#include \"${FILENAME}.h\"\n\n"
+    "char const* const ${NAME} = \n"
+    )
+
+  foreach(LINE ${LINES})
```
Comment (favreau) [inline CMake/StringifyPtx.cmake]: Done

File: plugins/engines/optix/cuda/CMakeLists.txt (hunk 2 above)
Comment (favreau) [inline plugins/engines/optix/cuda/CMakeLists.txt]: Done

File: plugins/engines/optix/cuda/CMakeLists.txt (hunk 1 above)
Comment (favreau) [inline plugins/engines/optix/cuda/CMakeLists.txt]: Done

File: plugins/engines/optix/cuda/CMakeLists.txt (hunk 3 above)
Comment (favreau) [inline plugins/engines/optix/cuda/CMakeLists.txt]: Done

**Pull request merged**